pkg/version/version.go

```diff
@@ -57,3 +58,40 @@ func Get() apimachineryversion.Info {
 		Platform:     fmt.Sprintf("%s/%s", runtime.GOOS, runtime.GOARCH),
 	}
 }
+
+// GetStableVersion returns the a stable version. In case the given version is a dev version then it returns the
+// last version. If it can't determine such a version it returns the given version again.
+func GetStableVersion(version string) (string, error) {
+	if !strings.Contains(version, "-dev") {
+		return version, nil
+	}
+
+	withVPrefix := strings.HasPrefix(version, "v")
+
+	v, err := semver.NewVersion(version)
+	if err != nil {
+		return "", err
+	}
+
+	if v.Patch() > 0 {
+		return constructVersion(withVPrefix, v.Major(), v.Minor(), v.Patch()-1), nil
+	}
+	if v.Minor() > 0 {
+		return constructVersion(withVPrefix, v.Major(), v.Minor()-1, v.Patch()), nil
+	}
+	if v.Major() > 0 {
+		return constructVersion(withVPrefix, v.Major()-1, v.Minor(), v.Patch()), nil
+	}
```

Comment:
As discussed offline, I would propose to calculate the `vx.y.z-dev-<commit>` image tag in case the running gardenlet version is a `-dev` version instead of this logic.